pytest_mergify/ci_insights.py

```diff
@@ -270,3 +374,54 @@ def mark_test_as_quarantined_if_needed(self, item: _pytest.nodes.Item) -> bool:
             return True
 
         return False
+
+
+def _select_affordable_tests(
+    budget_duration: datetime.timedelta,
+    test_durations: typing.Dict[str, datetime.timedelta],
+) -> typing.Dict[str, datetime.timedelta]:
+    """
+    Select tests that can be retried within the given budget.
+
+    This ensures we don't select tests that would exceed our time constraints
+    even with the minimum number of retries.
+    """
+
+    result = {}
+    for test, duration in test_durations.items():
+        if duration * _MIN_TEST_RETRY_COUNT <= budget_duration / len(test_durations):
```

Comment:
just to make it explicit 
```suggestion
        expected_retries_duration = duration * _MIN_TEST_RETRY_COUNT
        budget_per_test = budget_duration / len(test_durations):
        if expected_retries_duration <= budget_per_test:
```

Reply:
Will be solved with: https://linear.app/mergify/issue/MRGFY-5912/add-comments-to-the-allocate-test-retries-for-better-maintainability 👍